### PR TITLE
tetragon: Allow full exec path/args retrieval on 4.19 kernels

### PR DESCRIPTION
**Pull request by olsajiri**
To make it available we need to switch off the probe_arg_read
function that eats lots of instructions (~800).

With probe_arg_read gone there's enough space for the data
event interface to be compiled in. 

Signed-off-by: Jiri Olsa <jolsa@kernel.org>

### CODE REVIEW

File: bpf/process/data_event.h [hunk 1]
```diff
@@ -32,23 +36,31 @@ static long do_bytes(void *ctx, struct msg_data *msg, unsigned long arg,
...
+		return err;                                                    \
+	rd_bytes += err;                                                       \
+	if (rd_bytes == bytes)                                                 \
+		return 0;
+
+#define BYTES_COPY_5 BYTES_COPY BYTES_COPY BYTES_COPY BYTES_COPY BYTES_COPY
+
+	BYTES_COPY_5
+	BYTES_COPY_5
+#endif
```
Comment (kkourt) [inline bpf/process/data_event.h:64]: nit: maybe undef BYTES_COPY and BYTES_COPY_5 here?

**Review by kkourt — APPROVED** [review]
Thanks, looks good to me.

Just to ensure my understanding is correct: previous code used probe_arg_read() if the buffer was smaller than a certain size. With this patch, we removed it and we always use data_event_bytes. Removing it means that we now have enough instructions to use it for older 4.19 kernels as well.

File: bpf/process/data_event.h (hunk 1 above)
Comment (olsajiri) [reply bpf/process/data_event.h:64]: yes, will remove it, thanks

**Comment by olsajiri** [comment]
> Just to ensure my understanding is correct: previous code used probe_arg_read() if the buffer was smaller than a certain size. With this patch, we removed it and we always use data_event_bytes. Removing it means that we now have enough instructions to use it for older 4.19 kernels as well.

yes, that's correct

**Pull request merged**